src/main/java/com/gregtechceu/gtceu/utils/EntityDamageUtil.java

```diff
@@ -72,18 +72,15 @@ public static void applyFrostDamage(@NotNull LivingEntity entity, int damage) {
         if (damage <= 0) return;
         if (!entity.isAlive()) return;
         // snow/frost mobs cannot be chilled
-        if (entity instanceof SnowGolem || entity instanceof PolarBear || entity instanceof Stray)
+        if (entity.getType().is(EntityTypeTags.FREEZE_IMMUNE_ENTITY_TYPES))
             return;
         // frost walker entities cannot be chilled
         ItemStack stack = entity.getItemBySlot(EquipmentSlot.FEET);
         // check for empty in order to force damage to be applied if armor breaks
         if (!stack.isEmpty()) {
-            for (Tag base : stack.getEnchantmentTags()) {
-                CompoundTag compound = (CompoundTag) base;
-                if (compound.getShort("id") == FROST_WALKER_ID) {
-                    stack.hurtAndBreak(1, entity, ent -> ent.broadcastBreakEvent(EquipmentSlot.FEET));
-                    return;
-                }
+            if (stack.getEnchantmentLevel(Enchantments.FROST_WALKER) > 0) {
+                stack.hurtAndBreak(1, entity, ent -> ent.broadcastBreakEvent(EquipmentSlot.FEET));
+                return;
             }
         }
```

Comment:
Can't add a comment later but you forgor to use the chemical immune tag :trollface: